test/framework/clusterctl/clusterctl_helpers.go

```diff
@@ -123,6 +123,11 @@ type ApplyClusterTemplateAndWaitResult struct {
 // ApplyClusterTemplateAndWait gets a cluster template using clusterctl, and waits for the cluster to be ready.
 // Important! this method assumes the cluster uses a KubeadmControlPlane and MachineDeployments.
 func ApplyClusterTemplateAndWait(ctx context.Context, input ApplyClusterTemplateAndWaitInput, result *ApplyClusterTemplateAndWaitResult) {
+	ApplyClusterTemplateAndWaitWithArgs(ctx, input, result)
+}
+
+// ApplyClusterTemplateAndWaitWithArgs is similar to ApplyClusterTemplateAndWait except that it allows extra args to be passed during kubectl apply.
+func ApplyClusterTemplateAndWaitWithArgs(ctx context.Context, input ApplyClusterTemplateAndWaitInput, result *ApplyClusterTemplateAndWaitResult, args ...string) {
```

Comment:
What about adding args to the ApplyClusterTemplateAndWaitInput struct, so we are not required to change the signature?

Reply:
Yep, I think it's better to do it this way. Also I forgot to pass in the args to `ApplyWIthArgs` :) Will make the changes. Thanks!